### PR TITLE
Block Architecture Widget 

### PR DESCRIPTION
**Pull request by LeighFinegold**
## Summary

This PR documents the existing `block-architecture` widget implementation that provides **boundary-first architecture diagrams** using **Mermaid flowcharts**. The widget is already fully functional and supports the core requirements outlined in the original issue.

## Current Implementation ✅

The `block-architecture` widget already delivers:

### 1. **Boundary-First Architecture** 
- **Boundaries** created by `composed-of` (logical) and `deployed-in` (environmental) relationships
- **Children** appear inside their container boundaries 
- **Boundaries can nest** (systems containing subsystems)
- **Connections** (`connects`, `interacts`) drawn as edges that cross boundaries

### 2. **Mermaid Flowchart Output**
- Boundaries rendered as `subgraph`s with proper styling
- Leaf nodes as blocks positioned in their containers
- Edges from `connects`/`interacts` with relationship `description` as labels
- Optional highlighting with CSS classes
- Clickable links support via `link-prefix` and `link-map`

### 3. **Comprehensive Options API**

| Option | Type | Default | Description | Status |
|---|---|---:|---|---:|
| `focus-nodes` | string (comma delimited list) | — | Restrict view to these node IDs | ✅ Implemented |
| `focus-flows` | string (comma delimited list) | — | Restrict edges to flow transitions by unique-id/name | ✅ Implemented |
| `focus-controls` | string (comma delimited list) | — | Focus on nodes and relationships linked to these control IDs | ❌ Not implemented yet |
| `focus-interfaces` | string (comma delimited list) | — | Focus on nodes and relationships linked to these interface IDs | ❌ Not implemented yet |
| `focus-relationships` | string (comma delimited list) | — | Focus on nodes and relationships linked to these relationship IDs | ❌ Not implemented yet |
| `highlight-nodes` | string (comma delimited list) | — | Nodes to visually highlight | ✅ Implemented |
| `render-interfaces` | boolean | `false` | Show node interfaces as dotted attachments | ✅ Implemented |
| `include-containers` | `'none' \| 'parents' \| 'all'` | `'all'` | Which containers (systems) to draw | ✅ Implemented |
| `include-children` | `'none' \| 'direct' \| 'all'` | `'all'` | Include descendants when focusing containers | ✅ Implemented |
| `edges` | `'connected' \| 'seeded' \| 'all' \| 'none'` | `'connected'` | Edge inclusion strategy | ✅ Implemented |
| `node-types` | string (comma delimited list) | — | Filter by node-type | ✅ Implemented |
| `edge-labels` | `'description' \| 'none'` | `'description'` | Edge labeling strategy | ✅ Implemented |
| `link-prefix` | string | — | Prefix for clickable links | ✅ Implemented |
| `link-map` | stringified JSON | — | Explicit per-id links | ✅ Implemented |

## Test Fixtures Demonstrate Full Capability

The widget's comprehensive functionality is fully demonstrated in the test fixture suites across both modules:

### CALM Widgets Test Fixtures
Located in [`calm-widgets/test-fixtures/block-architecture-widget/`](./calm-widgets/test-fixtures/block-architecture-widget/):

- **[`basic-structures/`](./calm-widgets/test-fixtures/block-architecture-widget/basic-structures/)** - Core boundary-first architecture patterns
- **[`enterprise-bank-trading/`](./calm-widgets/test-fixtures/block-architecture-widget/enterprise-bank-trading/)** - Complex nested systems with trading flows  
- **[`focus-flows/`](./calm-widgets/test-fixtures/block-architecture-widget/focus-flows/)** - Flow-focused filtering and highlighting
- **[`interface-variations/`](./calm-widgets/test-fixtures/block-architecture-widget/interface-variations/)** - Interface rendering with dotted attachments
- **[`domain-interaction/`](./calm-widgets/test-fixtures/block-architecture-widget/domain-interaction/)** - Cross-boundary connections and interactions
- **[`complex-architecture/`](./calm-widgets/test-fixtures/block-architecture-widget/complex-architecture/)** - Large-scale system demonstrations
- **[`enterprise-bank-navigation/`](./calm-widgets/test-fixtures/block-architecture-widget/enterprise-bank-navigation/)** - Clickable link generation patterns
- **[`large-topology/`](./calm-widgets/test-fixtures/block-architecture-widget/large-topology/)** - Performance with extensive node networks

### CLI Test Fixtures
Located in [`cli/test_fixtures/template/widget-tests/`](./cli/test_fixtures/template/widget-tests/):

- **[`block-architecture-test.hbs`](./cli/test_fixtures/template/widget-tests/block-architecture-test.hbs)** - Block architecture widget usage examples and syntax patterns
- **[`sad-test.hbs`](./cli/test_fixtures/template/widget-tests/sad-test.hbs)** - Additional block architecture scenarios and edge cases

With corresponding expected outputs in [`cli/test_fixtures/template/expected-output/widget-tests/`](./cli/test_fixtures/template/expected-output/widget-tests/):

- **[`block-architecture-test.md`](./cli/test_fixtures/template/expected-output/widget-tests/block-architecture-test.md)** - Generated Mermaid diagrams from block-architecture-test.hbs
- **[`sad-test.md`](./cli/test_fixtures/template/expected-output/widget-tests/sad-test.md)** - Generated output from sad-test.hbs

## Syntax Examples

The CLI test fixtures demonstrate all supported syntax patterns in [`block-architecture-test.hbs`](./cli/test_fixtures/template/widget-tests/block-architecture-test.hbs):

**Full System View:**
```handlebars
{{block-architecture}}
```

**Flow-based View:**
```handlebars
{{block-architecture focus-flows="flow-conference-signup" highlight-nodes="attendees" }}
```

**Focus on Specific Nodes (no containers):**
```handlebars
{{block-architecture focus-nodes="load-balancer" include-containers="none"}}
```

**Full View with Interfaces:**
```handlebars
{{block-architecture render-interfaces=true}}
```
See output here [`block-architecture-test.md`](./cli/test_fixtures/template/expected-output/widget-tests/block-architecture-test.md)

### Immediate Actions
1. **Write unit tests** - Add comprehensive test coverage for the new focus options (`focus-controls`, `focus-interfaces`, `focus-relationships`)
2. **Confirm community acceptance** - Gather feedback from CALM community on the proposed options API and ensure general satisfaction with the current implementation

### Future Development
- Implement the remaining focus options for enhanced filtering capabilities
- Consider Cytoscape implementation + discuss if existing stuff can move to Mermaid.

## Conclusion

The `block-architecture` widget already provides a **production-ready, boundary-first architecture visualization solution** using Mermaid flowcharts. It supports all the core requirements from the original issue and demonstrates the power of the CALM widgets framework.
This establishes a solid foundation that can be extended with Cytoscape support and additional output formats as the project evolves.

---

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🎨 Code style/formatting changes
- [ ] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvements
- [ ] ✅ Test additions or updates
- [ ] 🔧 Chore (maintenance, dependencies, CI, etc.)

## Affected Components

- [ ] CLI (`cli/`)
- [ ] Shared (`shared/`)
- [x] CALM Widgets (`calm-widgets/`)
- [ ] CALM Hub (`calm-hub/`)
- [ ] CALM Hub UI (`calm-hub-ui/`)
- [x] Documentation (`docs/`)
- [ ] VS Code Extension (`calm-plugins/vscode/`)
- [ ] Dependencies
- [ ] CI/CD

## Testing

- [x] I have tested my changes locally
- [ ] I have added/updated unit tests
- [x] All existing tests pass

## Checklist
- [x] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [x] I have updated documentation if necessary
- [ ] I have added tests for my changes (if applicable)
- [x] My changes follow the project's coding standards

AI Notice
Copilot provided suggestions was used to improve clarity and efficiency and was used as an aid in generation of this PR description. The overall design and code remains developer-led.

### CODE REVIEW

**Comment by LeighFinegold** [comment]
Marked as Draft atm to get some initial feedback. If folks are happy, will implement unit tests and bring code coverage up to a suitable level.

File: calm-widgets/README.md [hunk 1]
```diff
@@ -124,9 +124,181 @@ classDef highlight fill:#f2bbae;
 ```
 
 
-5. **Test thoroughly** with `npm test`
```
Comment (markscott-ms) [inline calm-widgets/README.md:127]: We should be documenting more on the CALM website.

I believe most users of calm widgets won't have needed to clone the repo?

Comment (markscott-ms) [reply calm-widgets/README.md:127]: That's a future thought.

For now, this is beautiful. I've no issue with the option naming. Just add some code coverage like you have suggested.

**Review by markscott-ms — DISMISSED** [review]

File: calm-widgets/README.md (hunk 1 above)
Comment (LeighFinegold) [reply calm-widgets/README.md:127]: Will raise enhancement issues based on community feedback. (I suspect the community may want support for options, perhaps exclusion rules? )

Raised https://github.com/finos/architecture-as-code/issues/1599 for general calm.finos.org documentation updates.
Raised https://github.com/finos/architecture-as-code/issues/1600 to enhance the syntax for an interface scenario

**Review by rocketstack-matt — APPROVED** [review]
